src/Agent.Listener/MessageListener.cs

```diff
@@ -129,12 +146,15 @@ public sealed class MessageListener : AgentService, IMessageListener
             ArgUtil.NotNull(Session, nameof(Session));
             ArgUtil.NotNull(_settings, nameof(_settings));
             var agentServer = HostContext.GetService<IAgentServer>();
+            int consecutiveErrors = 0; //number of consecutive exceptions thrown by GetAgentMessageAsync
```

Comment:
same here, a bool make more sense, also better rename to something else, since you are not tracking how many consecutive error, you just check whether it is the first exception.